### ISSUE TITLE
Add support for UseNavigationViewWhenPossible

### ISSUE THREAD

**Issue by nmetulev** [Microsoft/WindowsTemplateStudio#1494]
## For new Features:

* **Overview:** With the 2.1 update of the UWP Community Toolkit, the HamburgerMenu now support NavigationView. However, the WTS templates will need to be updated to ensure the HamburgerMenu will look great when using (and not using) the NavigationView

* **Links:**
[HamburgerMenu NavigationView documentation](https://docs.microsoft.com/en-us/windows/uwpcommunitytoolkit/controls/HamburgerMenu#navview)
https://github.com/Microsoft/UWPCommunityToolkit/issues/1652

**Comment by mvegaca** [comment]
Working on it.

**Comment by mvegaca** [comment]
Hi @nmetulev 
I'm updating the tookit to 2.1.0 version. In case of Hamburguer menu I've seen that if it's on FCU W10 appears a header text when the window is in narrow state.

This is a screenshot of toolkit sample app.
![image](https://user-images.githubusercontent.com/5556784/33551915-4eeacb04-d8f3-11e7-9d41-b3a0d5c9f8d7.png)

How can I customize this text or avoid this behavior?

**Comment by mvegaca** [comment]
Could it be the NavigationView header that it isn't set to a HamburguerMenu property?

**Comment by nmetulev** [comment]
It's a known issue actually, here is the fix: https://github.com/Microsoft/UWPCommunityToolkit/pull/1655

**Comment by ralarcon** [comment]
Do you know when is planned to release this fix?

**Comment by nmetulev** [comment]
If you need this before February, we can do a 2.1.1 release to add this fix this month hopefully.

**Comment by crutkas** [comment]
@nmetulev we would need this before else it is a blocking issue.

**Comment by mvegaca** [comment]
I've been updating Toolkit version to 2.1.1 on [this issue branch](https://github.com/mvegaca/WindowsTemplateStudio/tree/Issue1494-navigationview) and adding conditional Xaml on window size change visual state changes.

I've found an app crash applying these changes to new Prism framework, It looks an unhandled exception when HamburgerControl sets the items source to NavigationView. I've been attached a crash repro [here](https://github.com/mvegaca/WindowsTemplateStudio/tree/Issue1494-navigationview/poc/HamburguerControl/PrismBugRepro).

@nmetulev can have a look on this? I think the we are adding the same code on different frameworks and it still crashing on Prism. Thank you much!

**Comment by crutkas** [comment]
@LeeParrishMSFT actually you may need to take a look from @mvegaca's comment above regarding Prism

**Comment by mvegaca** [comment]
Code added on PR #1590.
With this PR we can use UseNavigationViewWhenPossible on MVVMBasic, MVVMLight, CodeBehind, CaliburnMicro but isn't working on Prism

**Comment by ralarcon** [comment]
It looks there was an issue with the target framework on prism project.

**Comment by crutkas** [comment]
what changed to make it fail then work?

**Comment by ralarcon** [comment]
The target framework was not set to 10.0.16299.0, that makes the NavView rise the exception...

**Comment by crutkas** [comment]
check, so repro had an issue, not the toolkit.

**Comment by LeeParrishMSFT** [comment]
I also ran into a similar issue in that when the version of the UWP Community Toolkit was updated to the later version that the target runtime in the base Prism project is too old. It was still set to 5.2.3, and the other projects have come up to 6.0.1.

I ran into this while working on issue #1548 and noticed it when I started bringing the UWP Toolkit into that template that it was forcing a downgrade of the Toolkit due to the old runtime version. So I'm going to move up the target framework for the base Prism project to 6.0.1, but I'll need to run all the unit tests again to verify it doesn't introduce any other issues. We can also set the target version to 10.0.16299.0 and get everything updated to the match the latest versions for Prism.
